pkg/core/proxy/integrations/generic/decode.go

```diff
@@ -69,18 +69,19 @@ func decodeGeneric(ctx context.Context, logger *zap.Logger, reqBuf []byte, clien
 					logger.Debug("the genericRequests are:", zap.Any("h", string(genReq)))
 				}
 
-				reqBuffer, err := pUtil.PassThrough(ctx, logger, clientConn, dstCfg, genericRequests)
-				if err != nil {
-					utils.LogError(logger, err, "failed to passthrough the generic request")
-					return
-				}
-
-				genericRequests = [][]byte{}
-				logger.Debug("the request buffer after pass through in generic", zap.Any("buffer", string(reqBuffer)))
-				if len(reqBuffer) > 0 {
-					genericRequests = [][]byte{reqBuffer}
+				if opts.FallBackOnMiss {
+					reqBuffer, err := pUtil.PassThrough(ctx, logger, clientConn, dstCfg, genericRequests)
+					if err != nil {
+						utils.LogError(logger, err, "failed to passthrough the generic request")
+						return
+					}
+					genericRequests = [][]byte{}
+					logger.Debug("the request buffer after pass through in generic", zap.Any("buffer", string(reqBuffer)))
+					if len(reqBuffer) > 0 {
+						genericRequests = [][]byte{reqBuffer}
+					}
+					logger.Debug("the length of genericRequests after passThrough ", zap.Any("length", len(genericRequests)))
 				}
-				logger.Debug("the length of genericRequests after passThrough ", zap.Any("length", len(genericRequests)))
 				continue
```

Comment:
I don't think we should continue if we can't find any mocks and the passthrough is disabled. Could you check the behavior? You can do this by recording some generic mocks, deleting a few, and running in test mode. Ideally, we should only continue if the passthrough is successful. Otherwise, we should return an error due to the absence of mocks. This condition is valid for all the parsers.

Reply:
Yeah this should be done! On it!